frontend/src/fold/preprocess.js

```diff
@@ -51,3 +54,123 @@ export function moveRootFrameToFileFrames(foldModel) {
 	}
 	foldModel.file_frames = [firstFrame, ...foldModel.file_frames]
 }
+
+export function normalizeCoordinates(foldModel) {
+	const firstFrame = foldModel.file_frames[0]
+	const firstFrameVertices = firstFrame["vertices_coords"]
+
+	const firstFrameBoundingBox = computeBoundingBox(firstFrameVertices)
+	const firstFrameCenteredVertices = centerVertices(
+		firstFrameBoundingBox,
+		firstFrameVertices
+	)
+	const furthestDistanceFromOrigin = getFurthestDistanceFromOrigin(
+		firstFrameCenteredVertices
+	)
+
+	const normalizeFrame = (frame) => {
+		const vertices = frame["vertices_coords"]
+		if (!vertices || vertices.length === 0) {
+			return frame
+		}
+
+		const centeredCoords = centerVertices(firstFrameBoundingBox, vertices)
+		const normalized = normalizeBoundingBox(
+			centeredCoords,
+			DEFAULT_BOUNDING_BOX_DIAGONAL,
+			furthestDistanceFromOrigin
+		)
+		return {
+			...frame,
+			vertices_coords: normalized,
+		}
+	}
+	foldModel.file_frames = foldModel.file_frames.map(normalizeFrame)
+}
+
+function vectorLengthSquared(coordinates) {
+	const [x, y, maybeZ] = coordinates
+	const z = maybeZ || 0
+	return x * x + y * y + z * z
+}
+
+function getFurthestDistanceFromOrigin(vertices) {
+	if (!vertices || vertices.length === 0) {
+		return 0
+	}
+
+	const furthestPointFromOrigin = vertices.reduce((furthestYet, candidate) => {
+		if (
+			!furthestYet ||
+			vectorLengthSquared(candidate) > vectorLengthSquared(furthestYet)
```

Comment:
Hmm. Not a big deal at all, but there is a reason for not using reduce (except for the fact that the corresponding for loop would take the same amount of space). The reason is that with the for loop, if the 1st vertex is the furthest one, the `vectorLengthSquared` will be called only once for it (if we keep it as a variable), here it will be called each iteration.

Reply:
I agree, but I don't think it matters much. Sounds like a premature-optimization.